File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {

    const count = {};//숫자별 갯수를 세는 거니까 키값 형태로 저장하려고 등장횟수 객체 
    
    for(let num of array){
        count[num] = (count[num] !== undefined ? count[num] : 0) + 1;  
        //count[num] 이 undefined라면, count[num]을 등록하고, 등장횟수 1을 쓴다.
        //count[num]이 이미 있다면, +1을 해서 총 2가 된다.
    }
    
    //2. 가장 많이 나온 횟수 찾기
    let maxCount=0;
    for(let key in count){
        if(count[key] > maxCount){
            maxCount = count[key];
        }
    }
    
    //3.maxCount에 해당하는 숫자들을 배열로 모으기
    const mostFrequent=[];
    for(let key in count){
        if(count[key] === maxCount){
            mostFrequent.push(Number(key));
        }//key는 문자열이므로 숫자로 변환
    }
    
    //4.최빈값이 여러개면 -1, 하나면 그 값을 리턴
    if(mostFrequent.length >1) {
        return -1
    }else{
        return mostFrequent[0];
    }
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
       
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i = 0; i< array.length; i++){
            int key = array[i]; //인덱스 i번 째의 값이 키가 되는 것
            if(map.containsKey(key)){ //map에 key가 있으면 값이 한 번 더 나온거니까
                map.put(key, map.get(key)+1); // key의 값에 1을 더해줘라
            }else{ // 아니라면
                map.put(key, 1); // key: 0 -> key:1로 바꿔라
            }
        }
        int maxCount = 0; //가장 많이 나온 값
        int answer = 0; //최빈값 후보
        boolean multipleMax = false; // 최빈값이 2개 이상 있는지 확인하는 플래그
        
        for(int key : map.keySet()){ //키만 뽑아서
            int count = map.get(key); // 해당 숫자가 몇 번 나왔는지
            if(count > maxCount){
                maxCount = count;
                answer = key;
                multipleMax = false;
            }else if(count == maxCount){
                multipleMax = true;
            }
        }
        
        if(multipleMax) return -1;
        return answer;
    }
}